wpilibcIntegrationTests/src/main/native/cpp/DMATest.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <frc/AnalogInput.h>
+#include <frc/AnalogOutput.h>
+#include <frc/DigitalOutput.h>
+#include <frc/Jaguar.h>
+
+#include "TestBench.h"
+#include "frc/DMA.h"
+#include "frc/DMASample.h"
+#include "gtest/gtest.h"
+
+using namespace frc;
+
+static const double kDelayTime = 0.1;
+
+class DMATest : public testing::Test {
+ protected:
+  DMA* m_dma;
+  AnalogInput* m_analogInput;
+  AnalogOutput* m_analogOutput;
+  DigitalOutput* m_manualTrigger;
+  PWMSpeedController* m_pwm;
+
+  void SetUp() override {
+    m_analogOutput = new AnalogOutput(TestBench::kAnalogOutputChannel);
+    m_analogInput = new AnalogInput(TestBench::kFakeAnalogOutputChannel);
+    m_manualTrigger = new DigitalOutput(TestBench::kLoop1InputChannel);
+    m_pwm = new Jaguar(TestBench::kFakePwmOutput);
+    m_dma = new DMA();
+
+    m_dma->AddAnalogInput(m_analogInput);
+    m_dma->SetExternalTrigger(m_manualTrigger, false, true);
+    m_manualTrigger->Set(true);
+  }
+
+  void TearDown() override {
+    delete m_pwm;
+    delete m_manualTrigger;
+    delete m_analogOutput;
+    delete m_analogInput;
+    delete m_dma;
+  }
```

Comment:
Value types can be used here instead of pointers because the DMATest object's scope is limited to each TEST_F() block. A new DMATest object is created for each and destroyed at the end. Furthermore, the constructor can be used instead of SetUp().